include/swift/AST/DiagnosticsSema.def

```diff
@@ -825,6 +825,9 @@ ERROR(argument_out_of_order_unnamed_unnamed,none,
       "unnamed argument #%0 must precede unnamed argument #%1",
       (unsigned, unsigned))
 
+ERROR(func_is_not_an_instance_member,none,
+      "%0 with given arguments is not an instance member and "
+      "cannot be called using implicit 'self.'", (DeclName))
```

Comment:
IMO, since the implicit `self.` is implicit, it shouldn't be mentioned to the user (since they didn't write it). Something like `static member 'MyClass.max' shadows global function 'Swift.max'` might be more intuitive, especially if you could provide fix-it hints for inserting the `MyClass.` and/or `Swift.` (or whichever module the other candidate came from).